### ISSUE TITLE
Agda sits on a solution for a meta and does not apply it!?

### ISSUE THREAD

**Issue by ghost** [agda/agda#2093]
```
open import Data.Vec
open import Data.Nat
open import Data.Maybe

test : ∀ {n} -> Maybe (Vec ℕ n) -> Maybe (Vec ℕ n)
test (just x) = foldr (λ x -> {!Maybe ?!} ) (λ x -> λ { (just acc) -> just (x ∷ acc)
                                                      ; nothing -> nothing }) (just []) x
test nothing = nothing
```

---

I am using Agda 2.5.1 and stdlib 0.12

When refining the hole, this message is shown:
An internal error has occurred. Please report this as a bug.
Location of the error: src/full/Agda/TypeChecking/Reduce/Monad.hs:176

**Comment by ghost** [comment]
#2028 not sure if this is related or not

**Comment by asr** [comment]
> I am using Agda 2.5.1 and stdlib 0.12

After removing the extra `;` I couldn't reproduce the error. I got  the message `Cannot refine`.

**Comment by ghost** [comment]
there is no extra ;, the ; is for the pattern matching lambda

**Comment by asr** [comment]
Thanks for fixing the OP.

I could reproduce the problem with 2.5.1, 2.5.1.1 and stable-2.5.

**Comment by andreasabel** [comment]
The yellow around the hole should already warn you!  However, you should not get an internal error.
As one can see with `C-c C-=`, the hole `?0` is already solved with meta `_17 x`.  This might trigger the internal error.

**Comment by andreasabel** [comment]
In the light of #2028, it might be rather the postponed type checking problem

```
_18 := λ { (just acc) → just (x ∷ acc) ; nothing → nothing } :? _17 .n₁ → _17 (suc .n₁)
```

which causes the internal error.

Funnily, Agda sits on the solution for the hole, but does not apply it:

```
[16] (_17 .n) =< (Maybe (Vec ℕ .n)) : Set
```

**Comment by andreasabel** [comment]
The issue is fixed (see #2028), but I am still curious why Agda does not solve this meta...

**Comment by UlfNorell** [comment]
With `--show-implicits` the mystery is solved:

``` agda
[16] (_17 {.n} .n) =< (Maybe {lzero} (Vec {lzero} ℕ .n)) : Set
```
